packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -30,12 +30,16 @@ const defaultProps = {
 
 const DropdownItem = ({ className, children, isHovered, component: Component, isDisabled, ...props }) => {
   const additionalProps = props;
-  if (Component === 'a') {
+  if (children && children.type && children.type.name === 'Link') {
+    Component = 'div';
+    additionalProps.href = null;
+  } else if (Component === 'a') {
     additionalProps['aria-disabled'] = isDisabled;
     additionalProps.tabIndex = isDisabled ? -1 : additionalProps.tabIndex;
   } else if (Component === 'button') {
     additionalProps.disabled = isDisabled;
   }
+
   return (
     <li>
       <Component
```

Comment:
If there was another routing component with a different name then we'd have to modify this code again.
Also, would you need that extra div for react router link? I think we could change this to something like
```
// Check if we have a component, like react router's Link.
{React.isValidElement(children) ? (
  <li>{children}</li>
) : (
  <li><Component {...props} className=...>{children}</Component></li>
)}
```

Reply:
Good call here adding the isValidElement check.